cpp/src/prims/detail/nbr_intersection.cuh

```diff
@@ -397,26 +435,76 @@ struct copy_intersecting_nbrs_and_update_intersection_size_t {
       local_degree1 =
         static_cast<edge_t>(second_element_offsets[idx + 1] - second_element_offsets[idx]);
       indices1 = second_element_indices.begin() + second_element_offsets[idx];
+
+      local_edge_offset1 = second_element_offsets[idx];
+    }
+
+    if constexpr (std::is_same_v<SecondElementToIdxMap, void*>) {
+      if constexpr (!std::is_same_v<EdgeProperty_t, thrust::nullopt_t>) {
+        property1 = edge_partition_e_value_input.value_first() + local_edge_offset1;
+      }
+    } else {
+      if constexpr (!std::is_same_v<EdgeProperty_t, thrust::nullopt_t>) {
+        property1 = second_element_properties.begin() + local_edge_offset1;
+      }
     }
 
     // FIXME: this can lead to thread-divergence with a mix of high-degree and low-degree
     // vertices in a single warp (better optimize if this becomes a performance
     // bottleneck)
 
-    auto it =
-      thrust::set_intersection(thrust::seq,
-                               indices0,
-                               indices0 + local_degree0,
-                               indices1,
-                               indices1 + local_degree1,
-                               nbr_intersection_indices.begin() + nbr_intersection_offsets[i]);
+    auto inbr_start = nbr_intersection_indices.begin() + nbr_intersection_offsets[i];
+
+    auto it = thrust::set_intersection(thrust::seq,
+                                       indices0,
+                                       indices0 + local_degree0,
+                                       indices1,
+                                       indices1 + local_degree1,
+                                       inbr_start);
     thrust::fill(thrust::seq,
                  it,
                  nbr_intersection_indices.begin() + nbr_intersection_offsets[i + 1],
                  invalid_id);
 
-    return static_cast<size_t>(
-      thrust::distance(nbr_intersection_indices.begin() + nbr_intersection_offsets[i], it));
+    auto insection_size = static_cast<size_t>(thrust::distance(inbr_start, it));
+    if constexpr (!std::is_same_v<EdgeProperty_t, thrust::nullopt_t>) {
+      auto ip0_start = nbr_intersection_properties0.begin() + nbr_intersection_offsets[i];
+
+      // copy edge properties from first vertex to common neighbors
+      thrust::lower_bound(thrust::seq,
+                          indices0,
+                          indices0 + local_degree0,
+                          inbr_start,
+                          it,
+                          ip0_start,  // indices
+                          thrust::less<int>());
```

Comment:
Will this going to work for arbitrary `edge_property_value_t`? What if `edge_property_value_t` is bool or std::byte? And the index overflow?